### PR TITLE
Apalis imx8 spl conversion

### PR DESCRIPTION
**Pull request by igoropaniuk**
Please do not submit a Pull Request via github.  Our project makes use of
mailing lists for patch submission and review.  For more details please
see https://www.denx.de/wiki/U-Boot/Patches

### CODE REVIEW

**Review by MrCry0 — CHANGES_REQUESTED** [review]
I would split this commit into 3 parts:
- make required nodes available in SPL.
- fix SPL serial output issue.
- fix including the *u-boot.dtsi file to keep the platform dts file closer to the kernel variant.

**Review by Tim-Anderson — APPROVED** [review]
LGTM

**Review by vanmaegima — APPROVED** [review]

**Comment by ricardosalveti** [comment]
> I would split this commit into 3 parts:
> 
> * make required nodes available in SPL.
> * fix SPL serial output issue.
> * fix including the *u-boot.dtsi file to keep the platform dts file closer to the kernel variant.

I would just add that we might want to split the SPL enablement patch into 2 as well, one for the ifdef logic and another just for the config changes, as it is then easier to rebase changes in the future.

**Comment by igoropaniuk** [comment]
all comments addressed

**Review by MrCry0 — COMMENTED** [review]
Could you please swap **"enable SPL build"** and **"add support"** commits? It would have seemed more natural.
4aedff9 <-> 5575019

**Review by ricardosalveti — APPROVED** [review]
LGTM, I would just change the FIO tag to be 'toup' instead of 'internal' as you said we plan on upstreaming the changes.

File: configs/apalis-imx8_defconfig [hunk 1]
```diff
@@ -1,16 +1,31 @@
...
 CONFIG_ARCH_IMX8=y
+CONFIG_SPL_GPIO_SUPPORT=y
+CONFIG_SPL_LIBCOMMON_SUPPORT=y
+CONFIG_SPL_LIBGENERIC_SUPPORT=y
 CONFIG_SYS_TEXT_BASE=0x80020000
 CONFIG_SYS_MALLOC_F_LEN=0x8000
 CONFIG_ENV_SIZE=0x2000
 CONFIG_ENV_OFFSET=0xFFFFDE00
 CONFIG_DM_GPIO=y
+CONFIG_SPL_LOAD_IMX_CONTAINER=y
```
Comment (mike-scott) [inline configs/apalis-imx8_defconfig:13]: Are we sure we need this?
```
config SPL_LOAD_IMX_CONTAINER
        bool "Enable SPL loading U-Boot as a i.MX Container image"
        depends on SPL
        help
          This is to let SPL could load i.MX8 Container image
```

**Review by mike-scott — CHANGES_REQUESTED** [review]

File: board/toradex/apalis-imx8/apalis-imx8.c [hunk 2]
```diff
@@ -238,7 +239,9 @@ static void select_dt_from_module_version(void)
 			break;
 	}
 }
+#endif
 
+#ifndef CONFIG_SPL_BUILD
```
Comment (mike-scott) [inline board/toradex/apalis-imx8/apalis-imx8.c]: Seems like we can drop both of these additions, which would let the `#ifndef` block continue?

File: board/toradex/apalis-imx8/apalis-imx8.c [hunk 3]
```diff
@@ -268,7 +272,7 @@ int board_init(void)
 	return 0;
 }
 /* todo: With that function in ther is no console output in linux, drop for now */
-#if 0
+#ifdef CONFIG_SPL_BUILD
```
Comment (mike-scott) [inline board/toradex/apalis-imx8/apalis-imx8.c:277]: We might want to fix the todo comment above to reflect the new adjustment (as it's not entirely dropped now):
- We release the UART in the SPL hand-off
- But don't release it do to bug in the u-boot proper hand off

File: configs/apalis-imx8_defconfig (hunk 1 above)
Comment (mike-scott) [reply configs/apalis-imx8_defconfig:13]: (Discussion from Slack: this is the "default" defconfig.  We disable this for LmP where we use u-boot.itb)

**Review by mike-scott — CHANGES_REQUESTED** [review]

**Comment by mike-scott** [comment]
Looks good.  I only have 2 minor nits.

File: include/configs/apalis-imx8.h [hunk 4]
```diff
@@ -9,14 +9,44 @@
...
+ * 0x08081000 - 0x08180FFF is for m4_0 xip image,
+ * 0x08181000 - 0x008280FFF is for m4_1 xip image
+  * So 3rd container image may start from 0x8281000
+ */
+#define CONFIG_SYS_UBOOT_BASE 0x08281000
+#define CONFIG_SYS_MMCSD_FS_BOOT_PARTITION             0
+
+#define CONFIG_SPL_LDSCRIPT		"arch/arm/cpu/armv8/u-boot-spl.lds"
+#define CONFIG_SPL_STACK		0x013fff0
+#define CONFIG_SPL_BSS_START_ADDR      0x00130000
```
Comment (mike-scott) [inline include/configs/apalis-imx8.h]: Are these tabs or spaces?   This whole block of #define look a bit messy.

Comment (igoropaniuk) [reply include/configs/apalis-imx8.h]: fixed

**Review by mike-scott — APPROVED** [review]
LGTM

**Pull request merged**